x/emissions/metrics/metrics_util.go

```diff
@@ -0,0 +1,199 @@
+package metrics
```

Comment:
if this was all pulled from somewhere, you should add comments of linksk to all sources for future devs so they can better understand the context and provenance of the design decisions used here (and where it was borrowed from). Also, attribution is important for egos and sometimes legal reasons

Reply:
@kpeluso can you own or hand off looking into the licensing here?  who is owning this PR atm?